### PR TITLE
fixes #6824 - English default if browser locale in unsupported language

### PR DESCRIPTION
**Pull request by cfouant**

### CODE REVIEW

**Comment by domcleal** [comment]
Making English the first language seems like a fine default, but I'm just wondering if the bug report indicates an issue with the Rails controller that generates that text.  We made a change in 5a864bed in the API base controller as it had a similar issue - the locale wasn't being initialised, so it was showing text in random languages.  Adding that before_filter (we also have :set_gettext_locale_db in ApplicationController) ensured it was initialised properly.  Perhaps worth checking - so at least the locale will be followed if it is given.

File: lib/foreman/gettext/support.rb [hunk 1]
```diff
@@ -27,7 +27,11 @@ def self.register_available_locales locale_domain, locale_dir
           if Rails.env.test?
             FastGettext.default_available_locales = ['en']
           else
-            FastGettext.default_available_locales = Dir.glob(locale_search_path).collect {|f| locale_search_re.match(f)[1] }
+            locales = Dir.glob(locale_search_path).collect {|f|locale_search_re.match(f)[1] }
+
+            # FastGettext.default_available_locales will default to index at 0 if browser locale
+            # is set to an unsupported language, so must ensure that "en" is at head
+            FastGettext.default_available_locales = locales.unshift(locales.delete_at(locales.index("en")))
```
Comment (domcleal) [inline lib/foreman/gettext/support.rb]: nitpick: `locales.delete('en')` should do the same as the delete_at/index

Comment (cfouant) [inline lib/foreman/gettext/support.rb]: @domcleal totally valid nitpick, all fixed

**Comment by cfouant** [comment]
I was considering just explicitly setting the default locale to English, but shied away from it as I figured it would have been if it were intended.  I'm totally happy to make that fix.  I actually tested it this way, and it does not interfere with directly setting language or if browser locale is in a supported language, so it could readily work that way if you want to go that route.

**Comment by domcleal** [comment]
Sorry, I'm not sure what you mean by that.  I was wondering if the set_gettext_locale filter is missing from the controller that you're seeing this on - if so, that should definitely be fixed too, or you won't get any gettext initialisation.

**Comment by cfouant** [comment]
@domcleal - So, that was the problem.  I needed to add that filter.  Now it's fixed, without needing to alter anything in foreman.

**Comment by lzap** [comment]
I am wondering if we can make this explicit, or raise an error when locale is not set. This random behavior looks weird and we may hit this again. But I am not sure how to prevent from this. Maybe to check for the initialization in the `_` methods...

**Pull request closed without merging**